src/main/java/com/project/Investment/App/generatingModule/GeneratorPosition.java

```diff
@@ -0,0 +1,199 @@
+package com.project.Investment.App.generatingModule;
+
+import com.project.Investment.App.model.Position;
+import com.project.Investment.App.model.embeddedId.PositionId;
+import com.project.Investment.App.util.math.BigFunctions;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import java.time.LocalDate;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.counting;
+
+@Component
+@Slf4j
+public class GeneratorPosition {
+
+    public List<Position> generatorPosition(LocalDate startDate, String entityId, int total) {
+        List<Position> positions = new ArrayList<>();
+        Map<Integer, Integer> securityIdAndAggregateID = generatorSecurityIdAndAggregateID();
+        Double[] StartBmv = getDistributionBmv(securityIdAndAggregateID, total);
+        LocalDate finishDate = startDate.plusMonths(3);
+
+        for (int i = 0, securityId = 1; i < securityIdAndAggregateID.size(); i++, securityId++) {
+
+            Double bmvGross = StartBmv[i];
+            Double emvGross = getEmvGross(bmvGross);
+            Position position = Position.builder()
+                    .positionId(new PositionId(
+                            entityId,
+                            startDate,
+                            securityId,
+                            securityIdAndAggregateID.get(securityId)))
+                    .frequency('D')
+                    .weight(getWeight(bmvGross, (double) total))
+                    .bmvGross(bmvGross)
+                    .emvGross(emvGross)
+                    .grossReturn(getReturnGross(bmvGross, emvGross))
+                    .gainLossGross(getGainLossGross(emvGross, bmvGross))
+                    .build();
+            positions.add(position);
+        }
+
+        for (LocalDate date = startDate.plusDays(1); date.isBefore(finishDate); date = date.plusDays(1)) {
+
+            if (date.getDayOfWeek().getValue() < 6) {
```

Comment:
This line is not very friendly for a dev who's gonna support this code
You can extract it to a method `isBusinessDay` or add appropriate comment
Personally, I prefer the first option

Reply:
Done